src/input-helper.ts

```diff
@@ -89,6 +89,13 @@ export async function getInputs(): Promise<IGitSourceSettings> {
   }
   core.debug(`fetch depth = ${result.fetchDepth}`)
 
+  // Fetch jobs
```

Comment:
I think this logic should be moved below `line 114`

then we can do something like: 
~~~typescript
if(result.submodules){
    result.fetchJobs = Math.floor(Number(core.getInput('fetch-jobs') || '-1'))
    if (isNaN(result.fetchJobs) || result.fetchJobs < -1) {
        etc...
}
~~~

Reply:
Applied in d46e61f